plugins/accessibility/index.js

```diff
@@ -32,76 +57,187 @@ var AUDIT_FILE = path.join(__dirname, '../../node_modules/accessibility-develope
  */
 function teardown(config) {
 
+  var audits = [];
+
   if (config.chromeA11YDevTools) {
+    audits.push(runChromeDevTools(config));
+  }
+  // check for Tenon config and an actual API key, not the placeholder
+  if (config.tenonIO && /[A-Za-z][0-9]/.test(config.tenonIO.options.key)) {
+    audits.push(runTenonIO(config));
+  }
+  return q.all(audits).then(function() {
+    return outputResults();
+  });
+}
 
-    var data = fs.readFileSync(AUDIT_FILE, 'utf-8');
-    data = data + ' return axs.Audit.run();';
+var testOut = {failedCount: 0, specResults: []};
+var entities = new Entities();
 
-    var testOut = {failedCount: 0, specResults: []},
-        elementPromises = [];
+/**
+ * Audits page source against the Tenon API, if configured. Requires an API key:
+ * more information about licensing and configuration available at
+ * http://tenon.io/documentation/overview.php.
+ *
+ * @param {Object} config The configuration file for the accessibility plugin
+ * @return {q.Promise} A promise which resolves to the results of any passed or
+ *    failed tests
+ * @private
+ */
+function runTenonIO(config) {
+  var options = config.tenonIO.options;
+
+  return browser.driver.getPageSource().then(function(source) {
+    // setup response as a deferred promise
+    var deferred = q.defer();
+    request.post({
+      url: TENON_URL,
+      form: {
```

Comment:
As above, can we make all options available by just setting: options.src = source and the passing the whole thing here?